### PR TITLE
feature: Allow us to set how uploaded data should be interpreted.

### PR DESCRIPTION
**Pull request by Ctilhs**
As I mentioned in #44, it will allow us to set argument `value_input_opt` in update() function. So we can decide how data which we are sending will be interpreted.

### CODE REVIEW

File: df2gspread/df2gspread.py [hunk 1]
```diff
@@ -123,15 +125,15 @@ def upload(df, gfile="/New Spreadsheet", wks_name=None,
         cell_list = wks.range('%s%s:%s%s' % (first_col, start_row, last_col, start_row))
         for idx, cell in enumerate(cell_list):
             cell.value = df.columns.astype(str)[idx]
```
Comment (maybelinot) [inline df2gspread/df2gspread.py]: Current behavior force conversion to string of all values that goes to gspread, you would need to turn it off.

File: df2gspread/df2gspread.py [hunk 2]
```diff
@@ -123,15 +125,15 @@ def upload(df, gfile="/New Spreadsheet", wks_name=None,
...
-        wks.update_cells(cell_list)
+        wks.update_cells(cell_list, value_input_option=value_input_opt)
 
     # Addition of row names
     if row_names:
         cell_list = wks.range('%s%s:%s%d' % (
             start_col, first_row, start_col, last_idx))
         for idx, cell in enumerate(cell_list):
             cell.value = df.index.astype(str)[idx]
```
Comment (maybelinot) [inline df2gspread/df2gspread.py]: ditto

File: df2gspread/df2gspread.py [hunk 3]
```diff
@@ -123,15 +125,15 @@ def upload(df, gfile="/New Spreadsheet", wks_name=None,
...
-        wks.update_cells(cell_list)
+        wks.update_cells(cell_list, value_input_option=value_input_opt)
 
 
     # convert df values to string
```
Comment (maybelinot) [inline df2gspread/df2gspread.py]: ditto

**Comment by Ctilhs** [comment]
@maybelinot
- I think my change don't really affect rest of module. The default value is as it was before, now we have possibility to manipulate one more argument which is provided by google api, that's all. I mean sure it's affect the way how spreedsheet interpret input.
- But still, for sure some test would be a good thing.
- I use df2gspread with my modification and module from next point, I have not had any problems so far.
- I came across the problem with format of uploaded data, and from my observation, converting everyting to string works fine. To be sure that cells where we insert data have format that we want to I used  [gspread-formatting](https://github.com/robin900/gspread-formatting). I feel that it is more like work around than solution but it works.

I will try to figure something out.

**Comment by cancan101** [comment]
any chance of getting this merged?

File: df2gspread/df2gspread.py (hunk 1 above)
Comment (cancan101) [reply df2gspread/df2gspread.py:127]: what do you mean?

Comment (maybelinot) [reply df2gspread/df2gspread.py:127]: I mean that right now there is this `df.columns.astype(str)` trick

Comment (cancan101) [reply df2gspread/df2gspread.py:127]: Is that the case? That wasn't how I read the docs here: https://developers.google.com/sheets/api/reference/rest/v4/ValueInputOption

where I assumed strings would still be sent, google would then attempt to smart parse the string in the case of `USER_ENTERED`.

**Pull request closed without merging**